### PR TITLE
System tests: tighten 'is' operator

### PR DESCRIPTION
**Pull request by edsantiago**
Fix day-one sloppiness: when I first wrote this framework
it compared strings using 'expr', not '=', to be more
forgiving of extra cruft in output. This was a bad decision.
It means that warnings or additional text are ignored:

    is "all is ok, NOT!"  "all is ok"  <-- this would pass

Solution: tighten up the 'is' check. Use '=' (direct
compare) first. If it fails, look for wild cards ('*')
or character classes ('[') in the expect string. If
so, and only then, use 'expr'. And, thanks to a clever
suggestion from Luap99, include '(using expr)' in the
error message when we do so; this could make it easier
for a developer to understand a string mismatch.

This change exposes a lot of instances in which we weren't
doing proper comparisons. Fix those. Thankfully, there
weren't as many as I'd feared.

Also, and completely unrelated, add '-T' flag to bats
helper, for showing timing results. (I will open this
as a separate PR if requested. I too find it offensive
to jumble together unrelated commits.)

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

**Comment by edsantiago** [comment]
/hold

**DO NOT MERGE**. This cannot merge until #11775 is fixed! That's a serious nasty flake, and I bet this PR will hit it on most if not all of the `podman-remote` system test runs. I am submitting because this is (IMHO) code complete and ready for review.

**Comment by edsantiago** [comment]
Well, phooey! There's a new flake:
```
lstat /sys/fs/cgroup/devices/machine.slice/libpod-sha.scope: no such file or directory"'
```
I'll worry about it tomorrow.

**Comment by edsantiago** [comment]
And [yet another different one](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6422964389281792/html/sys-podman-fedora-33-rootless-host.log.html):
```
level=warning msg="cannot toggle freezer: cgroups not configured for container"'
level=warning msg="lstat : no such file or directory"'
```

**Comment by Luap99** [comment]
Nice work, that should help my brain to parse the errors.
Also the fact that you found new flakes proves the necessity of this.

**Comment by edsantiago** [comment]
This is blocked on #11775, #11784, and #11785.

**Comment by giuseppe** [comment]
#11784 is an issue in runc.  I suggest we just skip the test with runc and move on

**Comment by giuseppe** [comment]
and probably also #10442

File: test/system/005-info.bats [hunk 1]
```diff
@@ -37,7 +37,7 @@ cgroupVersion: v[12]
     # FIXME: if we're ever able to get package versions on Debian,
     #        add '-[0-9]' to all '*.package' queries below.
     tests="
-host.buildahVersion       | [0-9.]
+host.buildahVersion       | [0-9.]\\\+
```
Comment (edsantiago) [inline test/system/005-info.bats:40]: This will now fail if we ever vendor a `-dev` buildah, or something with letters.

File: test/system/125-import.bats [hunk 2]
```diff
@@ -20,26 +20,26 @@ load helpers
     # Simple import
     run_podman import -q $archive
     iid="$output"
-    run_podman run -t --rm $iid cat /random.txt
+    run_podman run --rm $iid cat /random.txt
```
Comment (edsantiago) [inline test/system/125-import.bats:23]: `-t` is tested in 450-interactive.bats (below), so I see little need to jump through the carriage-return hoops here. Please advise if there's something I've missed.

File: test/system/150-login.bats [hunk 3]
```diff
@@ -245,7 +245,7 @@ function _test_skopeo_credential_sharing() {
     is "$status" "0" "skopeo inspect - exit status"
 
     got_name=$(jq -r .Name <<<"$output")
-    is "$got_name" "$registry/$dest_name" "skopeo inspect -> Name"
+    is "$got_name" "$registry/$destname" "skopeo inspect -> Name"
```
Comment (edsantiago) [inline test/system/150-login.bats:248]: Oops. My bad. This has been broken from the beginning. This is a great example of why we need this PR.

File: test/system/160-volumes.bats [hunk 4]
```diff
@@ -261,7 +261,8 @@ EOF
 
     # prune should remove v4
     run_podman volume prune --force
-    is "$output" "${v[4]}" "volume prune, with 1, 2, 3 in use, deletes only 4"
+    is "$(echo $(sort <<<$output))" "${v[4]} ${v[5]} ${v[6]}" \
+       "volume prune, with 1, 2, 3 in use, deletes only 4, 5, 6"
```
Comment (edsantiago) [inline test/system/160-volumes.bats:265]: This is another great example of why we need this PR. Between the time I wrote this test and today, someone added some new volumes, and this check was not catching the addition.

File: test/system/700-play.bats [hunk 5]
```diff
@@ -30,7 +30,7 @@ spec:
   containers:
   - command:
     - sleep
-    - "100"
+    - \"100\"
```
Comment (edsantiago) [inline test/system/700-play.bats:33]: This and line 52 are completely unrelated; it's just something that makes color highlighting work better in my editors.

**Review by edsantiago — COMMENTED** [review]
@containers/podman-maintainers I believe this is ready to merge. Thank you @rhatdan and @giuseppe for your work on the blockers.

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons
thx again for the gr8 work @edsantiago !

File: test/system/005-info.bats (hunk 1 above)
Comment (rhatdan) [reply test/system/005-info.bats:40]: We regularly do this during development, will this block a merge to upstream?

Comment (edsantiago) [reply test/system/005-info.bats:40]: It probably would. Let me think of how to handle that; I will pore through git logs, find patterns, and resubmit. Thanks for letting me know.

**Comment by edsantiago** [comment]
/hold
the -dev thing is much more tedious than I had expected

**Comment by edsantiago** [comment]
Ready again.

If there are in-flight PRs which add BATS tests that do lax checking, and they merge without rebasing, we may end up with a broken main. I've looked through the top half of open PRs and see none, but I might have missed one.

**Review by vrothberg — COMMENTED** [review]
/lgtm
